datatypes/src/collections/feature_collection.rs

```diff
@@ -731,35 +731,42 @@ impl<'a, GeometryRef> FeatureCollectionRow<'a, GeometryRef> {
     }
 }
 
-pub struct FeatureCollectionIterator<'a, T> {
-    geometries: T,
+pub struct FeatureCollectionIterator<'a, GeometryIter> {
+    geometries: GeometryIter,
     time_intervals: std::slice::Iter<'a, TimeInterval>,
     data: Rc<HashMap<String, FeatureDataRef<'a>>>,
     row_num: usize,
 }
 
-impl<'a, T, GeometryRef> FeatureCollectionIterator<'a, T>
+impl<'a, GeometryIter, GeometryRef> FeatureCollectionIterator<'a, GeometryIter>
 where
-    T: std::iter::Iterator<Item = GeometryRef>,
+    GeometryIter: std::iter::Iterator<Item = GeometryRef>,
     GeometryRef: crate::primitives::GeometryRef,
 {
-    pub fn new(
-        geometries: T,
-        time_intervals: std::slice::Iter<'a, TimeInterval>,
-        data: Rc<HashMap<String, FeatureDataRef<'a>>>,
+    // unwrap can never throw because reserved columns are filtered previously
+    #![allow(clippy::missing_panics_doc)]
```

Comment:
You shouldn't describe it here, but rather replace `unwrap` with `expect` and use the string for ensurance.

Reply:
0e91839